File: app/src/main/java/com/hbjpro/androidvoiceassistant/Command/CommandBuilder.java
package com.hbjpro.androidvoiceassistant.Command;

import com.hbjpro.androidvoiceassistant.Tools.Tools;

import java.util.HashMap;

// More than one word for one order, next version

public class CommandBuilder {
    protected HashMap<String, Tools.OrderTy> _mapOrder;

    public CommandBuilder(){
        this._mapOrder = new HashMap<>();
    }

    /* --- Public Methods --- */
    public void createMapOrder(Tools.LanguageCode languageCode){
        switch (languageCode){
            case ENGLISH_AMERICA: _mapOrder = createMapEnglish(); break;
            case SPANISH: _mapOrder = createMapSpanish(); break;
            case JAPANESE: _mapOrder = createMapJapanese(); break;
            default: break;
        }
    }

    /* --- Protected Methods --- */
    protected Tools.OrderTy findOrder(String order){
        return _mapOrder.containsKey(order) ? _mapOrder.get(order) : Tools.OrderTy.INVALID;
    }

    /* --- Private Methods --- */
    private HashMap<String, Tools.OrderTy> createMapEnglish(){
        HashMap<String, Tools.OrderTy> map = new HashMap<>();
        map.put("open", Tools.OrderTy.OPEN_APP);
        map.put("search", Tools.OrderTy.SEARCH_INTERNET);

        return map;
    }
    private HashMap<String, Tools.OrderTy> createMapSpanish(){
        HashMap<String, Tools.OrderTy> map = new HashMap<>();
        map.put("abrir", Tools.OrderTy.OPEN_APP);
        map.put("busca", Tools.OrderTy.SEARCH_INTERNET);

        return map;
    }
    private HashMap<String, Tools.OrderTy> createMapJapanese(){
        HashMap<String, Tools.OrderTy> map = new HashMap<>();

        return map;
    }
}

File: app/src/main/java/com/hbjpro/androidvoiceassistant/Command/CommandBuilder.kt
package com.hbjpro.androidvoiceassistant.Command

import com.hbjpro.androidvoiceassistant.Tools.Tools

// More than one word for one order, next version

open class CommandBuilder {
    var _mapOrder:HashMap<String, Tools.CommandTy>

    init {
        _mapOrder = HashMap()
    }

    /* --- Public Methods --- */
    fun createMapOrder(languageCode: Tools.LanguageCode){
        when(languageCode){
            Tools.LanguageCode.ENGLISH_AMERICA -> _mapOrder = createMapEnglish()
            Tools.LanguageCode.SPANISH -> _mapOrder = createMapSpanish()
            Tools.LanguageCode.JAPANESE -> _mapOrder = createMapJapanese()
        }
    }

    /* --- Protected Methods --- */
    fun findOrder(order:String):Tools.CommandTy{
        return _mapOrder.getOrElse(order, {Tools.CommandTy.INVALID} )
    }

    /* --- Private Methods --- */
    private fun createMapEnglish():HashMap<String, Tools.CommandTy>{
        var map = HashMap<String, Tools.CommandTy>()
        map.put("open", Tools.CommandTy.OPEN_APP)
        map.put("search", Tools.CommandTy.SEARCH_INTERNET)

        return map
    }

    private fun createMapSpanish():HashMap<String, Tools.CommandTy>{
        var map = HashMap<String, Tools.CommandTy>()
        map.put("abrir", Tools.CommandTy.OPEN_APP)
        map.put("busca", Tools.CommandTy.SEARCH_INTERNET)

        return map
    }

    private fun createMapJapanese():HashMap<String, Tools.CommandTy>{
        var map = HashMap<String, Tools.CommandTy>()

        return map
    }
}